### PR TITLE
feat(components): spinner

### PR DESCRIPTION
**Pull request by dahaca**
## Description

* add a new `Spinner` component

## Related Issue

Resolve https://github.com/trezor/trezor-suite/issues/9019

### CODE REVIEW

File: packages/components/src/components/loaders/Spinner/Spinner.tsx [hunk 1]
```diff
@@ -1,51 +1,78 @@
...
+    const onLoopComplete = useMemo(
+        () => () => {
+            if (!hasFinished || !hasError) {
+                return;
+            }
+
+            setHasFinishedRotation(true);
+        },
+        [hasFinished, hasError],
+    );
```
Comment (komret) [inline packages/components/src/components/loaders/Spinner/Spinner.tsx]: You could `useCallback`. However, I am not sure if the memoization is helping anything here.

File: packages/components/src/components/loaders/Spinner/Spinner.tsx [hunk 2]
```diff
@@ -1,51 +1,78 @@
...
+                />
+            ) : (
+                <StyledLottie
+                    animationData={animationStart}
+                    onComplete={() => setHasStarted(true)}
+                    loop={false}
+                    {...commonProps}
+                />
+            )}
+        </>
```
Comment (komret) [inline packages/components/src/components/loaders/Spinner/Spinner.tsx]: Better approach IMO (refactored by Copilot 🤖 ):

```ts
    const getProps = () => {
        if (hasFinished && hasFinishedRotation) {
            return {
                animationData: animationEnd,
                loop: false,
            };
        }
        if (hasError && hasFinishedRotation) {
            return {
                animationData: animationWarn,
                loop: false,
            };
        }
        if (hasStarted || !isWithSpeadup) {
            return {
                animationData: animationMiddle,
                onLoopComplete,
            };
        }
        return {
            animationData: animationStart,
            onComplete: () => setHasStarted(true),
            loop: false,
        };
    };

    return <StyledLottie size={size} isGrey={isGrey} className={className} {...getProps()} />;
```

**Review by komret — CHANGES_REQUESTED** [review]
Beautiful! Some nitpicks:

File: packages/components/src/components/loaders/Spinner/Spinner.tsx (hunk 1 above)
Comment (dahaca) [reply packages/components/src/components/loaders/Spinner/Spinner.tsx]: b3f3c48bf5

File: packages/components/src/components/loaders/Spinner/Spinner.tsx (hunk 2 above)
Comment (dahaca) [reply packages/components/src/components/loaders/Spinner/Spinner.tsx]: b3f3c48bf5

**Comment by dahaca** [comment]
2 commits are there by accident btw, will squash.

**Review by komret — APPROVED** [review]

**Pull request merged**